core/types.ml

```diff
@@ -1995,28 +1999,144 @@ module Policy = struct
+    let all_opts = [PresenceOmit ; AliasOmit ; ArrowsShowTheOneEffect ; ArrowsCurriedCollectionAssumeFresh ; ContractOperationArrows ; OpenDefault ; DifferentOperationArrows ]
+
+    let show_opt : opt -> string
+      = function
+      | PresenceOmit             -> "presence_omit"
+      | ArrowsShowTheOneEffect   -> "arrows_show_the_one_effect"
+      | ArrowsCurriedCollectionAssumeFresh -> "arrows_curried_collection_assume_fresh"
+      | AliasOmit                -> "alias_omit"
+      | ContractOperationArrows  -> "contract_operation_arrows"
+      | OpenDefault              -> "open_default"
+      | DifferentOperationArrows -> "different_operation_arrows"
+    let string_of_opts = Settings.string_of_paths -<- List.map show_opt
+
+    let parse_opts : string -> opt list
+      = let parse_opt : string -> opt
+          = fun s ->
+          match String.lowercase_ascii s with
+          | "presence_omit" | "pres"
+            -> PresenceOmit
+          | "arrows_show_the_one_effect"
+            -> ArrowsShowTheOneEffect
+          | "arrows_curried_collection_assume_fresh" | "ccf"
+            -> ArrowsCurriedCollectionAssumeFresh
+          | "alias_omit" | "alias"
+            -> AliasOmit
+          | "contract_operation_arrows" | "contract"
+            -> ContractOperationArrows
+          | "open_default" | "|.}"
+            -> OpenDefault
+          | "different_operation_arrows" | "->>"
+            -> DifferentOperationArrows
+          | _ -> failwith ("Invalid option: " ^ s)
+        in
+        let is_correct : opt list -> bool
+          = not -<- ListUtils.has_duplicates
+        in
+        fun s ->
+        match String.lowercase_ascii s with
+        | "none"    -> []
+        | "default" -> default_opts
+        | "all"     -> all_opts
+        | _ -> let lst = List.map parse_opt (Settings.parse_paths s) in
+               if is_correct lst then lst
+               else failwith "Options cannot be duplicated."
+
+    let syno
+      = let fst = "Fine grained control over effect sugar (only works when \
+                   effect_sugar = true)." in
+        let lines =
+          [  "Options (shortcuts in [brackets]):"
+           ; " * presence_omit [pres]: omit presence polymorphic operations"
+           ; "   within effect rows (1)"
+           ; " * alias_omit [alias]: hide empty (1) shared effect rows in last"
+           ; "   argument of aliases"
+           ; " * arrows_show_the_one_effect: display the imlicit shared effect"
+           ; "   on arrows"
+           ; " * arrows_curried_collection_assume_fresh [ccf]: in curried"
+           ; "   functions, collection arrows are assumed to have fresh"
+           ; "   effects and these are hidden"
+           ; " * contract_operation_arrows [contract]: contract operations"
+           ; "   `E:() {}-> a' to `E:a'"
+           ; " * open_default [|.}]: effect rows are open by default,"
+           ; "   closed with syntax { |.}"
```

Comment:
Have you implemented the closing syntax?

Reply:
I don't see it as part of this PR.

Reply:
It is implemented on the printer side, but not in the parser. I'll disable it for now (keeping the code, I'll just make the option unaccessible), and then reenable it when parser support is ready.

Reply:
I have added the closing syntax into this PR (patched the Lexer, Parser and the desugaring pass).